// Copyright 2024 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
// All generated protos should be opaque, with "xxx_hidden_" prepended to their field names.


package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";


enum ComputeAlphaPacketMirroringFilterDirectionEnum {
  ComputeAlphaPacketMirroringFilterDirectionEnumNO_VALUE_DO_NOT_USE = 0;
  ComputeAlphaPacketMirroringFilterDirectionEnumINGRESS = 1;
  ComputeAlphaPacketMirroringFilterDirectionEnumEGRESS = 2;
}

enum ComputeAlphaPacketMirroringEnableEnum {
  ComputeAlphaPacketMirroringEnableEnumNO_VALUE_DO_NOT_USE = 0;
  ComputeAlphaPacketMirroringEnableEnumTRUE = 1;
  ComputeAlphaPacketMirroringEnableEnumFALSE = 2;
}

message ComputeAlphaPacketMirroring {
  int64 id = 1;
  string self_link = 2;
  string name = 3;
  string description = 4;
  string region = 5;
  ComputeAlphaPacketMirroringNetwork network = 6;
  int64 priority = 7;
  ComputeAlphaPacketMirroringCollectorIlb collector_ilb = 8;
  ComputeAlphaPacketMirroringMirroredResources mirrored_resources = 9;
  ComputeAlphaPacketMirroringFilter filter = 10;
  ComputeAlphaPacketMirroringEnableEnum enable = 11;
  string project = 12;
  string location = 13;
}

message ComputeAlphaPacketMirroringNetwork {
  string url = 1;
  string canonical_url = 2;
}

message ComputeAlphaPacketMirroringCollectorIlb {
  string url = 1;
  string canonical_url = 2;
}

message ComputeAlphaPacketMirroringMirroredResources {
  repeated ComputeAlphaPacketMirroringMirroredResourcesSubnetworks subnetworks = 1;
  repeated ComputeAlphaPacketMirroringMirroredResourcesInstances instances = 2;
  repeated string tags = 3;
}

message ComputeAlphaPacketMirroringMirroredResourcesSubnetworks {
  string url = 1;
  string canonical_url = 2;
}

message ComputeAlphaPacketMirroringMirroredResourcesInstances {
  string url = 1;
  string canonical_url = 2;
}

message ComputeAlphaPacketMirroringFilter {
  repeated string cidr_ranges = 1;
  repeated string ip_protocols = 2;
  ComputeAlphaPacketMirroringFilterDirectionEnum direction = 3;
}

message ApplyComputeAlphaPacketMirroringRequest {
  ComputeAlphaPacketMirroring resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteComputeAlphaPacketMirroringRequest {
  string service_account_file = 1;
  ComputeAlphaPacketMirroring resource = 2;
}

message ListComputeAlphaPacketMirroringRequest {
  string service_account_file = 1;
  string Project = 2;
  string Location = 3;
}

message ListComputeAlphaPacketMirroringResponse {
  repeated ComputeAlphaPacketMirroring items = 1;
}

service ComputeAlphaPacketMirroringService {
  rpc ApplyComputeAlphaPacketMirroring(ApplyComputeAlphaPacketMirroringRequest) returns (ComputeAlphaPacketMirroring);
  rpc DeleteComputeAlphaPacketMirroring(DeleteComputeAlphaPacketMirroringRequest) returns (google.protobuf.Empty);
  rpc ListComputeAlphaPacketMirroring(ListComputeAlphaPacketMirroringRequest) returns (ListComputeAlphaPacketMirroringResponse);
}
